lib/rules/no-unused-prop-types.js

```diff
@@ -980,28 +509,31 @@ module.exports = {
     function handleFunctionLikeExpressions(node) {
       handleSetStateUpdater(node);
       markDestructuredFunctionArgumentsAsUsed(node);
-      markAnnotatedFunctionArgumentsAsDeclared(node);
+    }
+
+    function handleCustomValidators(component) {
+      const propTypes = component.declaredPropTypes;
```

Comment:
This piece of code deals with used propTypes, but it was nested inside of `markPropTypesAsDeclared` (which was extracted out).